src/extension/background-script/connectors/lawallet.ts

```diff
@@ -128,7 +128,7 @@ export default class LaWallet implements Connector {
     return {
       data: {
         transactions: parsedTransactions.sort(
-          (a, b) => b.settleDate - a.settleDate
+          (a, b) => (b.settleDate ?? 0) - (a.settleDate ?? 0)
```

Comment:
I don't think this connector should be affected like this in this PR? if lawallet only has settled transactions, there is now extra code that does not even apply.

Can't you sort the transactions before mapping them?

Reply:
that was just to make typescript happy we have common type definitions for all the connectors. now settleDate can be either number | null (cause of nwc pending and failed transactions). any better approach here?

Reply:
Can't you sort the transactions before mapping them?

Reply:
how this will solve the problem of settledDate being  number | null. and avoid fallback value. its the type that is why we need fallback here. but it will never being triggered as all other transactions will returned settled transactions only

Reply:
because the settleDate cannot be null in the lawallet connector response. Only after you map it does it become nullable.

try sorting the `data` field.